src/Payum/Paypal/Rest/PaypalRestGatewayFactory.php

```diff
@@ -35,21 +35,30 @@ protected function populateConfig(ArrayObject $config)
                 'client_id' => '',
                 'client_secret' => '',
                 'config_path' => '',
+                'config' => [],
             ];
             $config->defaults($config['payum.default_options']);
 
-            $config['payum.required_options'] = ['client_id', 'client_secret', 'config_path'];
+            $config['payum.required_options'] = ['client_id', 'client_secret'];
             $config['payum.api'] = function (ArrayObject $config) {
                 $config->validateNotEmpty($config['payum.required_options']);
 
-                if (false == defined('PP_CONFIG_PATH')) {
-                    define('PP_CONFIG_PATH', $config['config_path']);
-                } elseif (PP_CONFIG_PATH !== $config['config_path']) {
-                    throw new InvalidArgumentException(sprintf('Given "config_path" is invalid. Should be equal to the defined "PP_CONFIG_PATH": %s.', PP_CONFIG_PATH));
+                if (isset($config['config_path']) && $config['config_path'] !== '') {
+                    if (false == defined('PP_CONFIG_PATH')) {
+                        define('PP_CONFIG_PATH', $config['config_path']);
+                    } elseif (PP_CONFIG_PATH !== $config['config_path']) {
+                        throw new InvalidArgumentException(sprintf('Given "config_path" is invalid. Should be equal to the defined "PP_CONFIG_PATH": %s.', PP_CONFIG_PATH));
+                    }
                 }
 
                 $credential = new OAuthTokenCredential($config['client_id'], $config['client_secret']);
-                return new ApiContext($credential);
+                $apiContext = new ApiContext($credential);
+
+                if (isset($config['config']) && is_array($config['config'])) {
```

Comment:
I think this will always evaluate to `true` since you are already setting the config option to an empty array in the default options. So maybe this check can just be removed, or maybe you can rather check that the config array is not empty?

Reply:
Yes, you're right, it'll always evaluate to `true` - actually, after looking at the `\PayPal\Rest\ApiContext::setConfig` source code once again, I think it's fine to just remove that check.